server/src/main/scala/services/Crunch.scala

```diff
@@ -352,24 +358,21 @@ object RunnableCrunchGraph {
 
   import akka.stream.scaladsl.GraphDSL.Implicits._
 
-  def apply(
-             flightsSource: Source[Flights, _],
-             voyageManifestsSource: Source[Set[VoyageManifest], _],
-             cruncher: CrunchGraphStage,
-             crunchStateActor: ActorRef): RunnableGraph[NotUsed] =
-    RunnableGraph.fromGraph(GraphDSL.create() { implicit builder =>
-      val crunchSink = Sink.actorRef(crunchStateActor, "completed")
-      val F: Outlet[Flights] = builder.add(flightsSource).out
-      val M: Outlet[Set[VoyageManifest]] = builder.add(voyageManifestsSource).out
+  def apply[M](
+                flightsSource: Source[Flights, M],
+                voyageManifestsSource: Source[VoyageManifests, M],
+                cruncher: CrunchGraphStage,
+                crunchStateActor: ActorRef): RunnableGraph[(M, M, NotUsed, NotUsed)] = {
+    val crunchSink = Sink.actorRef(crunchStateActor, "completed")
 
-      val CR = builder.add(cruncher)
-      val G = builder.add(crunchSink)
+    RunnableGraph.fromGraph(GraphDSL.create(flightsSource, voyageManifestsSource, cruncher, crunchSink)((_, _, _, _)) { implicit builder =>
+      (fs, ms, cr, cs) =>
```

Comment:
you are gonna have to explain this part to me next week.

Reply:
will do! it's a vital change for testability but hidden away in the docs